dfm/description/gen_description.py

```diff
@@ -0,0 +1,247 @@
+
+from collections import defaultdict
+
+from spacy.language import Language
+from spacy.matcher import Matcher
+from spacy.tokens import Doc
+import spacy
+
+sys.path.append(".")
+from dfm.description.description_pattern_lists import (
+    female_gendered_terms,
+    male_gendered_terms,
+    occupation_pattern_list,
+    danish_adult_words,
+)
+
+
+def remove_irrelevant_columns(ds):
+    return ds.remove_columns(["text", "doc_id", "LICENSE", "uri", "date_built"])
+
+
+def get_muslim_name_patterns() -> list:
+    from dacy.datasets import muslim_names
+
+    muslim_names_list = [name.lower() for name in muslim_names()["first_name"]]
+
+    return MatchCounter.term_list_to_lowercase_match_patterns(
+        term_list=muslim_names_list, label="muslim_names"
+    )
+
+
+def get_gender_name_patterns() -> list:
+    """
+    Creates a list of SpaCy patterns in the shape {"[col_name]": [{"LOWER": "[pattern]"}]}
+    """
+    from dacy.datasets import female_names, male_names
+
+    female_names_list = [name.lower() for name in female_names()["first_name"]]
+    female_names_patterns = MatchCounter.term_list_to_lowercase_match_patterns(
+        female_names_list, label="female_names"
+    )
+
+    male_names_list = [name.lower() for name in male_names()["first_name"]]
+    male_name_patterns = MatchCounter.term_list_to_lowercase_match_patterns(
+        male_names_list, label="male_names"
+    )
+
+    return female_names_patterns + male_name_patterns
+
+
+class MatchCounter:
+    """Class of utility functions for counting spacy matches"""
+
+    def __init__(self, match_patterns: list, nlp: Language):
+        self.nlp = nlp
+        self.matcher_objects = self.gen_matcher_objects_from_pattern_list(match_patterns)
+
+    @staticmethod
+    def term_list_to_lowercase_match_patterns(
+        term_list: list, label: Optional[str] = None, label_prefix: str = ""
+    ) -> list:
+        """
+        Takes a list of terms and creates a list of SpaCy patterns in the shape {"label": [{"LOWER": "term"}]}
+        """
+        out_list = []
+
+        for term in term_list:
+            if label is None:
+                cur_label = label_prefix + term
+                out_list.append({cur_label: [{"LOWER": term}]})
+            else:
+                cur_label = label_prefix + label
+                out_list.append({cur_label: [{"LOWER": term}]})
+
+        return out_list
+
+    def gen_matcher_objects_from_pattern_list(
+        self, pattern_container_list: list
+    ) -> Matcher:
+        """
+        Generates matcher objects from a list of dictionaries with {matcher_label (str): pattern (list)}
+        Pattern must conform to SpaCy pattern standards:
+
+        Example input:
+            >>> pattern_container_list = [
+            >>>    {"atheism": [{"LOWER": {"REGEX": "athei.+"}}]},
+            >>>    {"atheism": [{"LOWER": {"REGEX": "atei.+"}}]},
+            >>>    {"skøde": [{"LOWER": "skøde"}]},
+            >>> ]
```

Comment:
The default way to write an example is using Example or Examples

```suggestion
        Example:
            >>> pattern_container_list = [
            >>>    {"atheism": [{"LOWER": {"REGEX": "athei.+"}}]},
            >>>    {"atheism": [{"LOWER": {"REGEX": "atei.+"}}]},
            >>>    {"skøde": [{"LOWER": "skøde"}]},
            >>> ]
            >>> match_counter.gen_matcher_objects_from_pattern_list(pattern_container_list)
```